### PR TITLE
PLAT-481: Allow custom partition keys

### PR DESCRIPTION
**Pull request by mwstobo**
@ukayani

### CODE REVIEW

**Comment by Melraidin** [comment]
:+1:

**Comment by mwstobo** [comment]
Thanks Kevin!

File: lib/px/service/kinesis/base_request.rb [hunk 1]
```diff
@@ -90,11 +90,12 @@ def flush_records
     # The data will be msgpacked and queued for send.
     #
     # Returns the number of unsent messages
-    def queue_record(data)
+    def queue_record(data, partition_key=nil)
```
Comment (ukayani) [inline lib/px/service/kinesis/base_request.rb:93]: 👍

**Review by ukayani — APPROVED** [review]
LGTM. Thanks

**Pull request merged**